File: codes/cpp/inheritance/single/student.cpp
#include "student.h"

Student::Student(std::string name, int grade):
    Person(name)
{
    this->grade = grade;
}

void Student::info()
{
    Person::info();
    std::cout << "My grade: " << this->grade << "\n";
    std::cout << "My luckyNumber2: " << this->luckyNumber << "\n";
}

File: codes/php/inheritance/single/student.php
<?php

require_once "person.php";

class Student extends Person {

    private $grade;

    public function __construct($name, $grade) {
        parent::__construct($name);
        $this->grade = $grade;
    }

    public function info(){
        parent::info();
        echo "My grade is: $this->grade\n";
    }

}

?>
